awx_collection/plugins/modules/tower_application.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/python
+# coding: utf-8 -*-
+
+# (c) 2017, Wayne Witzel III <wayne@riotousliving.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: tower_application
+author: "Geoffrey Bacheot (@jffz)"
+short_description: create, update, or destroy Ansible Tower applications
+description:
+    - Create, update, or destroy Ansible Tower applications. See
+      U(https://www.ansible.com/tower) for an overview.
+options:
+    name:
+      description:
+        - Name of the application.
+      required: True
+      type: str
+    description:
+      description:
+        - Description of the application.
+      type: str
+    authorization_grant_type:
+      description:
+        - The grant type the user must use for acquire tokens for this application.
+      default: "password"
+      choices: ["password", "authorization-code"]
+      type: str
+      required: True
+    client_id:
+      description:
+        - Desired client_id for application. Self generated if empty, returned in response
+      type: str
+    # TODO it can be defined but is never returned through api, need to check
+    client_secret:
+      description:
+        - Desired client_secret for application. Self generated, returned in response
+      type: str
+    client_type:
+      description:
+        - Set to public or confidential depending on how secure the client device is.
+      choices: ["public", "confidential"]
+      type: str
+      required: True
+    organization:
+      description:
+        - Name of organization for application.
+      type: str
+      required: True
+    redirect_uris:
+      description:
+        - Allowed urls list, space separated. Required when authorization-grant-type=authorization-code
+      default: "present"
+      choices: ["present", "absent"]
+      type: str
+    state:
+      description:
+        - Desired state of the resource.
+      default: "present"
+      choices: ["present", "absent"]
+      type: str
+    host:
```

Comment:
@jffz @AlanCoding I'm pulling this into a role for `redhat_cop/tower_configuration`, and want to make sure I'm mapping correctly in the task.  Should host, username, and password be here?